### ISSUE TITLE
The new extjs() robustness makes it difficult to cancel running tools.

### ISSUE THREAD

**Issue by cmillion** [cmillion/gPhoton#178]
There should be some kind of catch in there for this. Because it's really inconvenient when you start a long job only to realize a second after you hit <ENTER> that it wasn't quite what you wanted. Which I do all the time.

**Comment by bshiao** [comment]
Hmm, I’m not sure I understand. If you have a long running job and you ctrl-c or kill it shortly after starting, or for that matter at any time during the course of the job, only the currently running DB query would continue to run and eventually removed from cache or killed because a timeout occurred because no client issued that same query again.

**Comment by cmillion** [comment]
Yeah. It's not a database-side issue. In my zeal to create more robust query handling in manage_requests2(), I put in a loop that silently caches errors and tries again (up to, like, 100 consecutive attempts) unless it gets the ERROR response from the server. So just hitting Ctrl-C does nothing. So I need to figure out a way to specifically catch Ctrl-C so that it exits the loop. The issue is just here to remind me.
